sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1070,6 +1107,7 @@ public Instant getWatermark() {
 
     @Override
     public CheckpointMark getCheckpointMark() {
+      reportBacklog();
```

Comment:
Yes and no.
No because it is a public API and runners can call it as part of their flow in a scope within metrics could be reported.
Yes because runners are not required to call this method as part of their flow.